### PR TITLE
Updated pretty.py with newest commits from the IPython project

### PR DESCRIPTION
**Pull request by melund**
The pretty printer we are using from ipython project has some advanced features. So it is easy to fix if there are any of the built-in structures in xonsh that doesn't display nicely. So if you have any particular issues I can try to fix it.

### CODE REVIEW

File: xonsh/pretty.py [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
+replace the old `pprint` python module which does not allow developers
+to provide their own pretty print callbacks.
 
 This module is based on ruby's `prettyprint.rb` library by `Tanaka Akira`.
 
-The following implementations were forked from the IPython project:
-* Copyright (c) 2008-2014, IPython Development Team
-* Copyright (C) 2001-2007 Fernando Perez <fperez@colorado.edu>
-* Copyright (c) 2001, Janko Hauser <jhauser@zscout.de>
-* Copyright (c) 2001, Nathaniel Gray <n8gray@caltech.edu>
```
Comment (scopatz) [inline xonsh/pretty.py]: The copyrights need to be maintained due to IPython's license.

File: xonsh/pretty.py [hunk 2]
```diff
@@ -83,6 +77,7 @@ def _repr_pretty_(self, p, cycle):
             Portions (c) 2009 by Robert Kern.
 :license: BSD License.
 """
+from __future__ import print_function
```
Comment (scopatz) [inline xonsh/pretty.py]: STrintly speaking, I don't think we need this :)

**Comment by scopatz** [comment]
Thanks for putting this in @melund. I think the copyright is the only big issue.

**Comment by melund** [comment]
Ups. I forgot about the copyright notice. I weakly remember adding it the first time we added pretty.py. It should be up fixed now.

**Comment by scopatz** [comment]
Looks good to me! Thanks @melund!

**Pull request merged**